firstuseauthenticator/firstuseauthenticator.py

```diff
@@ -6,15 +6,63 @@
 locally in a dbm file, and checked next time they log in.
 """
 import dbm
+import os
+from jinja2 import ChoiceLoader, FileSystemLoader
 from jupyterhub.auth import Authenticator
+from jupyterhub.handlers import BaseHandler
 from jupyterhub.orm import User
 
-from tornado import gen
+from tornado import gen, web
 from traitlets.traitlets import Unicode, Bool
 
 import bcrypt
 
 
+TEMPLATE_DIR = os.path.join(os.path.dirname(__file__), 'templates')
+
+
+class ResetPasswordHandler(BaseHandler):
+    """Render the reset password page."""
+    def __init__(self, *args, **kwargs):
+        self._loaded = False
+        super().__init__(*args, **kwargs)
+
+    def _register_template_path(self):
+        if self._loaded:
+            return
+
+        self.log.debug('Adding %s to template path', TEMPLATE_DIR)
+        loader = FileSystemLoader([TEMPLATE_DIR])
+
+        env = self.settings['jinja2_env']
+        previous_loader = env.loader
+        env.loader = ChoiceLoader([previous_loader, loader])
+
+        self._loaded = True
+
+    @web.authenticated
+    async def get(self):
+        self._register_template_path()
+        html = self.render_template('reset.html')
+        self.finish(html)
+
+    @web.authenticated
+    async def post(self):
+        data = {}
+        for arg in self.request.arguments:
+            data[arg] = self.get_argument(arg, strip=False)
+        user = self.get_current_user()
+        data['username'] = user.name
+        self.authenticator.reset_password(data)
+
+        html = self.render_template(
+            'reset.html',
+            result=True,
+            result_message='password changed successfully',
```

Comment:
Can you change the message to say `your password has been changed successfully'?